src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -214,36 +219,33 @@ class PaymentMethodList extends Component {
     render() {
         return (
             <>
-                <FlatList
-                    data={this.getFilteredPaymentMethods()}
-                    renderItem={this.renderItem}
-                    keyExtractor={item => item.key}
-                    ListEmptyComponent={this.renderListEmptyComponent()}
-                    ListHeaderComponent={this.props.listHeaderComponent}
-                />
-                {
-                    this.props.shouldShowAddPaymentMethodButton
-                    && (
-                        <FormAlertWrapper>
-                            {
-                                isOffline => (
-                                    <Button
-                                        text={this.props.translate('paymentMethodList.addPaymentMethod')}
-                                        icon={Expensicons.CreditCard}
-                                        onPress={e => this.props.onPress(e)}
-                                        isDisabled={this.props.isLoadingPayments || isOffline}
-                                        style={[styles.mh4, styles.buttonCTA]}
-                                        iconStyles={[styles.buttonCTAIcon]}
-                                        key="addPaymentMethodButton"
-                                        success
-                                        shouldShowRightIcon
-                                        large
-                                    />
-                                )
-                            }
-                        </FormAlertWrapper>
-                    )
-                }
+                {this.props.isLoadingPaymentMethods && !this.props.network.isOffline ? (
+                    <ScrollView>
+                        <ActivityIndicator color={themeColors.spinner} size="large" />
+                    </ScrollView>
+                ) : (
+                    <FlatList
+                        data={this.getFilteredPaymentMethods()}
+                        renderItem={this.renderItem}
+                        keyExtractor={item => item.key}
+                        ListEmptyComponent={this.renderListEmptyComponent()}
+                        ListHeaderComponent={this.props.listHeaderComponent}
+                    />
+                )}
+                {this.props.shouldShowAddPaymentMethodButton && (
+                    <Button
+                        text={this.props.translate('paymentMethodList.addPaymentMethod')}
+                        icon={Expensicons.CreditCard}
+                        onPress={e => this.props.onPress(e)}
+                        isDisabled={this.props.isLoadingPayments || this.props.network.isOffline}
```

Comment:
This code says that the button should be disable if loading the payments so that's why I'm confused about that.

Reply:
So maybe just change that condition to only consider isOffline?

Reply:
ohhh that's a good point